nemo_skills/pipeline/app.py

```diff
@@ -180,3 +180,145 @@ def add_mount_path(mount_source: str, mount_dest: str, cluster_config):
 
     else:
         raise ValueError("No mounts found in cluster config, can only add to existing mount list.")
+
+
+def is_mounted_filepath(filepath: str, cluster_config: dict):
+    """
+    Check if the filepath is mounted using the cluster config.
+
+    Args:
+        filepath: The filepath to check.
+        cluster_config: The cluster configuration dictionary.
+
+    Returns:
+        bool: Whether the filepath is mounted.
+    """
+    # Find which mount path matches the filepaths prefix
+    for mount in cluster_config['mounts']:
+        mount_source, mount_dest = mount.split(':')
+        if filepath.startswith(mount_dest):
+            return True
+
+    return False
+
+
+def get_mounted_filepath(filepath: str, cluster_config: dict):
+    """
+    Resolve the mounted filepath using the cluster config to merge the mount destination path to the filepath.
+    If an already mounted path is provided, then it will return the filepath as is.
+
+    Args:
+        filepath: The filepath to resolve.
+        cluster_config: The cluster configuration dictionary.
+
+    Returns:
+        The resolved filepath.
+    """
+    # Find which mount path matches the filepaths prefix
+    mount_path = None
+    for mount in cluster_config['mounts']:
+        mount_source, mount_dest = mount.split(':')
+        if filepath.startswith(mount_source):
+            mount_path = mount
+            break
+
+        elif filepath.startswith(mount_dest):
+            # already mounted, return immediately
+            return filepath
+
+    if mount_path is None:
+        raise ValueError(
+            f"Could not find a mount path for the file path `{filepath}`. Below paths are mounted: \n"
+            f"{cluster_config['mounts']}"
+        )
+
+    # replace the mount destination inside the filepath with the mount source
+    mount_source, mount_dest = mount_path.split(':')
+    filepath = mount_dest + filepath[len(mount_source) :]  # replace the mount destination with the mount source
+
+    return filepath
+
+
+def get_unmounted_filepath(filepath: str, cluster_config: dict,):
```

Comment:
is this the same as https://github.com/NVIDIA/NeMo-Skills/blob/main/nemo_skills/pipeline/utils.py#L114? Should we move all path utils in a separate file? Maybe we should have a utils folder and all these files there or something like this

Reply:
I can move these util functions into pipeline/utils.py. For this function specifically, it should be merged into above.

As to utils - there's two sets of utils that's commonly necessary - pipeline.cli and pipeline.utils. So we can copy parts of pipeline.utils into a another file maybe to actually be utils or just treat pipeline.utils as the generic functions to interact with pipeline.*?

Reply:
I just feel that our utils files are way too big and have too many functions. So if there is some reasonable way to split them it would be good to do that. Especially pipeline/utils.py. But we can do it in another PR